### PR TITLE
Compile for scala js and update to scala 2.13.1

### PR DESCRIPTION
**Pull request by benderpremier**
I want to create a UI in scala js to illustrate some of the aima book. I need to compile the code to scala js hence this PR.

I also took the occasion to update to scala `2.13.1`

### CODE REVIEW

File: core/src/main/scala/aima/core/agent/Actuator.scala [hunk 1]
```diff
@@ -5,15 +5,14 @@ import aima.core.random.Randomness
 /**
   * @author Shawn Garner
   */
-trait Actuator[Action, Percept] {
-  def agent: Agent[Action, Percept]
-  def act(action: Action, environment: Environment[Action, Percept]): Environment[Action, Percept]
+trait Actuator[E, A] {
+  def act(action: A, e: E): E //could be Action => E => E
```
Comment (BusyByte) [inline core/src/main/scala/aima/core/agent/Actuator.scala]: @benderpremier I feel like making it e:E  is just too generic. I can't figure out from looking at it what it means.  I would need a decoding manual to figure out what E represents.  I like it the way it was before.  Also we lost the notion of Percept and Environment and Action.  I don't think this will help learning or matching up things to the algorithms in the book.

Comment (BusyByte) [reply core/src/main/scala/aima/core/agent/Actuator.scala]: @benderpremier are all these changes needed just for scala 2.13 and scala js?  Could we actually do these in smaller separate PRs?

Comment (benderpremier) [reply core/src/main/scala/aima/core/agent/Actuator.scala]: Hi @BusyByte,

Sorry the last two commits weren't meant for this branch. I didn't pay attention when pushing it yesterday. Let me correct that.

Comment (benderpremier) [reply core/src/main/scala/aima/core/agent/Actuator.scala]: Done, it should be easier to review now. Sorry about that :)

File: core/src/test/scala/aima/core/environment/vacuum/ModelBasedReflexVacuumAgentSpec.scala [hunk 2]
```diff
@@ -57,7 +57,7 @@ class ModelBasedReflexVacuumAgentSpec extends Specification {
   }
 
   "should do nothing if low on power" in new context {
-    val resultStream = Stream.continually(agent.agentFunction.apply(DirtyPercept))
+    val resultStream = LazyList.continually(agent.agentFunction.apply(DirtyPercept))
```
Comment (BusyByte) [inline core/src/test/scala/aima/core/environment/vacuum/ModelBasedReflexVacuumAgentSpec.scala]: Why did you change this to a list.  I see that [Stream.continually still exists in scala 2.13](https://www.scala-lang.org/api/current/scala/collection/immutable/Stream$.html#continually[A](elem:=%3EA):scala.collection.immutable.Stream[A])

File: core/src/test/scala/aima/core/environment/vacuum/TableDrivenVacuumAgentSpec.scala [hunk 3]
```diff
@@ -66,7 +66,7 @@ class TableDrivenVacuumAgentSpec extends Specification {
 
   "table driven agent must persist all percepts" in new context {
     val rnd = new Random()
-    val randomPerceptStream: Stream[VacuumPercept] = Stream.continually {
+    val randomPerceptStream: LazyList[VacuumPercept] = LazyList.continually {
```
Comment (BusyByte) [inline core/src/test/scala/aima/core/environment/vacuum/TableDrivenVacuumAgentSpec.scala]: Same as above.

File: core/src/test/scala/aima/core/environment/vacuum/ModelBasedReflexVacuumAgentSpec.scala (hunk 2 above)
Comment (BusyByte) [reply core/src/test/scala/aima/core/environment/vacuum/ModelBasedReflexVacuumAgentSpec.scala]: Nevermind, I see it is deprecated now `(Since version 2.13.0) Use LazyList (which is fully lazy) instead of Stream (which has a lazy tail only)` . That is too bad I don't really care for the new name.
